File: backend/src/main/java/io/blog/devlog/domain/mail/service/EmailService.java
package io.blog.devlog.domain.mail.service;

import io.blog.devlog.domain.mail.types.MessageType;

import java.util.HashMap;

public interface EmailService {
    void sendEmail(MessageType type, String email, String subject, HashMap<String, String> params);
    String setContext(MessageType type, HashMap<String, String> params);
}

File: backend/mail-service/src/main/kotlin/io/blog/mail/mail/service/EmailService.kt
package io.blog.mail.mail.service

import io.blog.mail.mail.types.MessageType

interface EmailService {
    fun sendEmail(type: MessageType, email: String, subject: String, params: HashMap<String, String>)
    fun setContext(type: MessageType, params: HashMap<String, String>) : String
}